validator/client/src/main/java/tech/pegasys/teku/validator/client/loader/ValidatorLoader.java

```diff
@@ -45,25 +45,22 @@
 
   private final SlashingProtector slashingProtector;
   private final AsyncRunner asyncRunner;
-  private final Supplier<HttpClient> remoteValidatorHttpClientFactory;
+  private Supplier<HttpClient> externalValidatorHttpClientFactory;
 
-  @VisibleForTesting
-  ValidatorLoader(
-      final SlashingProtector slashingProtector,
-      final AsyncRunner asyncRunner,
-      final Supplier<HttpClient> remoteValidatorHttpClientFactory) {
+  private ValidatorLoader(
+      final SlashingProtector slashingProtector, final AsyncRunner asyncRunner) {
     this.slashingProtector = slashingProtector;
     this.asyncRunner = asyncRunner;
-    this.remoteValidatorHttpClientFactory = remoteValidatorHttpClientFactory;
   }
 
   public static ValidatorLoader create(
       final SlashingProtector slashingProtector, final AsyncRunner asyncRunner) {
-    return new ValidatorLoader(
-        slashingProtector,
-        asyncRunner,
-        Suppliers.memoize(
-            () -> HttpClient.newBuilder().version(HttpClient.Version.HTTP_1_1).build()));
+    return new ValidatorLoader(slashingProtector, asyncRunner);
+  }
+
+  @VisibleForTesting
+  void setExternalValidatorHttpClientFactory(final Supplier<HttpClient> httpClientSupplier) {
+    this.externalValidatorHttpClientFactory = httpClientSupplier;
   }
```

Comment:
i think having the supplier be final is probably a better pattern, than allowing it to be set just for testing.
Maybe what we want is a second 'create' function that allows the supplier to be provided during create time to allow for this testing? it could be package private, just used in testing...

Reply:
I've refactored httpclient factory to be loaded in initializeValidator method (instead of being a class member) ... and created an overloaded initializeValidator that can utilitze mocked httpclient in unit tests. The factory is only required (and will be initialized) if external public keys are defined.

Reply:
we can't really set httpClient up in constructor anymore because the keystore/truststore values are derived from "ValidatorConfig" object, which is injected with initializeValidator.

Reply:
ok fair enough...